File: distros/J/JavonetPerlSdk/lib/Javonet/Binaries/Python/javonet/utils/connectionData/WsConnectionData.py
import hashlib

from javonet.utils.ConnectionType import ConnectionType
from javonet.utils.connectionData.IConnectionData import IConnectionData


class WsConnectionData(IConnectionData):

    def __init__(self, hostname: str):
        self._hostname = hostname
    @property
    def connection_type(self):
        return ConnectionType.WebSocket

    @property
    def hostname(self):
        return self._hostname

    def serialize_connection_data(self):
        return [self.connection_type.value, 0, 0, 0, 0, 0, 0]

    def __eq__(self, other):
        return isinstance(other, WsConnectionData) and self._hostname == other.hostname

    def __hash__(self):
        return int(hashlib.sha1(f"{self.hostname}".encode()).hexdigest(), 16)

    @hostname.setter
    def hostname(self, value):
        self._hostname = value

File: distros/J/JavonetPerlSdk/lib/Javonet/Binaries/Nodejs/utils/connectionData/WsConnectionData.js
const ConnectionType = require('../../utils/ConnectionType');
const IConnectionData = require('./IConnectionData');

class WsConnectionData extends IConnectionData {
    constructor(hostname) {
        super();
        this._hostname = hostname;
        this._connectionType = ConnectionType.WEB_SOCKET;
    }

    get connectionType() {
        return this._connectionType;
    }

    get hostname() {
        return this._hostname;
    }

    set hostname(value) {
        this._hostname = value;
    }

    serializeConnectionData() {
        return [this.connectionType, 0, 0, 0, 0, 0, 0];
    }

    equals(other) {
        return other instanceof WsConnectionData && this._hostname === other.hostname;
    }

}

module.exports = WsConnectionData;